### DISCUSSION TITLE
how to restore changes due to git push force

### DISCUSSION THREAD

**Discussion by TabawaX** [community/community#146637] [New to GitHub] [answered]
how to restore changes because GitHub force?

**Comment by pooulad** [comment, marked as answer]
If GitHub has forced changes or you need to restore previous changes, you can recover them depending on the situation:

### 1\. **Restoring from Local Repository**

If the changes exist locally:

*   Use ```bash git reflog``` to find the commit hash of the previous state.
    
*   Run ```bash git reset --hard``` to revert the repository to that state.
    

### 2\. **Using GitHub's UI**

If the changes are in a branch or pull request:

*   Navigate to the specific branch or pull request in the GitHub repository.
    
*   Use the **History** section to view previous commits.
    
*   Create a new branch or revert directly from the UI using the "Revert" option available on specific commits.
    

### 3\. **Restoring Deleted Branches**

If a branch was deleted:

*   Check if the branch exists locally with git branch.
    
*   If it does, push it back to GitHub using git push origin .
    
*   If not, check the reflog (git reflog) for commits related to the branch.
    

### 4\. **Undoing Forced Changes from GitHub Actions**

If a GitHub Action or a collaborator force-pushed changes:

*   Use the repository's history to locate the state before the force-push.
    
*   Fetch the commit hash and reset or create a new branch to restore that state.
